compiler/rustc_errors/src/diagnostic.rs

```diff
@@ -502,6 +509,23 @@ impl Diagnostic {
         self
     }
 
+    /// Adds a suggestion intended only for a tool. The intent is that the metadata encode
```

Comment:
```suggestion
    /// Adds a suggestion intended only for a tool. The intent is that the metadata encodes
```

Reply:
<pedantry>
Technically "data" and hence "metadata" are plural, so "encode" is strictly correct.
</pedantry>
 But in common use we use data as a (singular) mass noun so "encodes" would be the right form to agree.